File: java/po/world_simulator/InputManager.java
package po.world_simulator;

import java.util.concurrent.LinkedBlockingQueue;

public class InputManager {
    private int lastInput = 0;
    private boolean newTurnPressed = false;
    private LinkedBlockingQueue<Integer> inputQueue = new LinkedBlockingQueue<>();

    public InputManager() {
    }

    public void pushInput(int input) {
        inputQueue.offer(input);
    }

    public void readNextInput() {
        newTurnPressed = false;
        try {
            // [AI] Blokowanie głównego wątku pętli aż GUI dorzuci klawisz do kolejki
            int input = inputQueue.take();

            if (input == Config.NEW_TURN) {
                newTurnPressed = true;
                return;
            }
            lastInput = input;
        } catch (InterruptedException e) {
            e.printStackTrace();
        }
    }

    public int getLastInput() {
        return this.lastInput;
    }

    public void clearLastInput() {
        this.lastInput = 0;
    }

    public boolean isQuitPressed() {
        return lastInput == Config.QUIT;
    }

    public boolean isNewTurnPressed() {
        return newTurnPressed;
    }
}

File: python/po/world_simulator/InputManager.py
import queue
from .Config import Config

class InputManager:
    def __init__(self):
        self.lastInput = 0
        self.newTurnPressed = False
        self.inputQueue = queue.Queue()


    def pushInput(self, input_val):
        self.inputQueue.put(input_val)


    def readNextInput(self):
        self.newTurnPressed = False

        if self.inputQueue.empty():
            return

        input_val = self.inputQueue.get(block=False)
        
        if input_val == Config.NEW_TURN:
            self.newTurnPressed = True
            return
        self.lastInput = input_val


    def getLastInput(self):
        return self.lastInput


    def clearLastInput(self):
        self.lastInput = 0


    def isQuitPressed(self):
        return self.lastInput == Config.QUIT


    def isNewTurnPressed(self):
        return self.newTurnPressed
